File: Task/Exceptions/Objective-C/exceptions-3.m
@try {
  [self foo];
}
@catch (MyException *exc) {
  //Catch only your specified type of exception
}
@catch (NSException *exc) {
  //Catch any NSException or subclass
  NSLog(@"caught exception named %@, with reason: %@", [exc name], [exc reason]);
}
@catch (id exc) {
  //Catch any kind of object
}
@finally {
  //This code is always executed after exiting the try block
}

File: Task/Exceptions/Java/exceptions-3.java
try {
   fooChecked();
}
catch(MyException exc) {
   //Catch only your specified type of exception
}
catch(Exception exc) {
   //Catch any non-system error exception
}
catch(Throwable exc) {
   //Catch everything including system errors (not recommended)
}
finally {
   //This code is always executed after exiting the try block
}
